### ISSUE TITLE
Warning checking for shadow-cljs deps can have false positives

### ISSUE THREAD

**Issue by frankitox** [liquidz/vim-iced#220]
Just something to keep in mind. You can have false positives of this warning if you use shadow-cljs with other tool. For example, if you use `deps.edn` to specify the [dependencies](https://liquidz.github.io/vim-iced/vim-iced.html#vim-iced-manual-shadow-cljs). Because the code only searches for the deps in `shadow-cljs.edn`. For example:

deps.edn
```
{:paths
 ["src"]

 :deps
 {reagent {:mvn/version "0.10.0"}
  thheller/shadow-cljs {:mvn/version "2.9.8"}}

 :aliases
 {:dev
  {:extra-deps ;; https://liquidz.github.io/vim-iced/vim-iced.html#vim-iced-manual-shadow-cljs
   {refactor-nrepl {:mvn/version "2.5.0"}
    cider/cider-nrepl {:mvn/version "0.24.0"}
    iced-nrepl {:mvn/version "1.0.1"}}}}}
```

shadow-cljs.edn
```
{:deps true

 ;; https://liquidz.github.io/vim-iced/vim-iced.html#vim-iced-manual-shadow-cljs
 :nrepl {:cider false
         :middleware [cider.nrepl/wrap-classpath
                      cider.nrepl/wrap-clojuredocs
                      cider.nrepl/wrap-complete
                      cider.nrepl/wrap-debug
                      cider.nrepl/wrap-format
                      cider.nrepl/wrap-info
                      cider.nrepl/wrap-macroexpand
                      cider.nrepl/wrap-ns
                      cider.nrepl/wrap-out
                      cider.nrepl/wrap-spec
                      cider.nrepl/wrap-test
                      cider.nrepl/wrap-trace
                      cider.nrepl/wrap-undef
                      cider.nrepl/wrap-xref
                      refactor-nrepl.middleware/wrap-refactor
                      iced.nrepl/wrap-iced]}

 :builds
 {:app
  {:target :react-native
   :init-fn app.core/init
   :output-dir "rn/app"
   :compiler-options {:infer-externs :auto}
   :devtools {:repl-pprint true}
   :js-options {:js-package-dirs ["rn/node_modules"]}}}}
```
And to start the shadow-cljs watcher
> shadow-cljs -A:dev watch app

**Comment by liquidz** [comment]
Thanks!! I'll have a look.

**Comment by liquidz** [comment]
@frankitox I rewrote a validation with babashka script.
Currently, only supports `deps.edn`.
https://shadow-cljs.github.io/docs/UsersGuide.html#deps-edn
`Leiningen` is a to-do.
https://shadow-cljs.github.io/docs/UsersGuide.html#Leiningen

Could you try `dev` branch?

**Comment by frankitox** [comment]
Hi @liquidz !

Well, the problem is I'm using `:deps true` instead of `:deps {:aliases ..`. But if I do add `:aliases [:dev]` to my config, it works. I think it's good enough. I mean, this is already a ton of code just to show a goddam warning.

Thank you!

**Comment by liquidz** [comment]
@frankitox Thanks for your confirmation!

vim-iced can't know what options are used when you start shadow-cljs.
So `:deps true` will only refer `:deps` values in your `deps.edn`.

**Comment by liquidz** [comment]
closed by #221
